src/ImageSharp.Web/ImageSharpRequestAuthorizationUtilities.cs

```diff
@@ -0,0 +1,288 @@
+        /// <param name="query">The query collection.</param>
+        /// <param name="handling">The command collection handling.</param>
+        /// <returns>The computed HMAC.</returns>
+        public Task<string> ComputeHMACAsync(HostString host, PathString path, QueryString queryString, QueryCollection query, CommandHandling handling)
+            => this.ComputeHMACAsync(this.ToHttpContext(host, path, queryString, query), handling);
+
+        /// <summary>
+        /// Compute a Hash-based Message Authentication Code (HMAC) for request authentication.
+        /// </summary>
+        /// <param name="context">The request HTTP context.</param>
+        /// <param name="handling">The command collection handling.</param>
+        /// <returns>The computed HMAC.</returns>
+        public string ComputeHMAC(HttpContext context, CommandHandling handling)
+            => AsyncHelper.RunSync(() => this.ComputeHMACAsync(context, handling));
+
+        /// <summary>
+        /// Compute a Hash-based Message Authentication Code (HMAC) for request authentication.
+        /// </summary>
+        /// <param name="context">The request HTTP context.</param>
+        /// <param name="handling">The command collection handling.</param>
+        /// <returns>The computed HMAC.</returns>
+        public async Task<string> ComputeHMACAsync(HttpContext context, CommandHandling handling)
+        {
+            byte[] secret = this.options.HMACSecretKey;
+            if (secret is null || secret.Length == 0)
+            {
+                return null;
+            }
+
+            CommandCollection commands = this.requestParser.ParseRequestCommands(context);
+            if (handling == CommandHandling.Sanitize)
+            {
+                this.StripUnknownCommands(commands);
+            }
+
+            ImageCommandContext imageCommandContext = new(context, commands, this.commandParser, this.parserCulture);
+            return await this.options.OnComputeHMACAsync(imageCommandContext, secret);
+        }
+
+        /// <summary>
+        /// Compute a Hash-based Message Authentication Code (HMAC) for request authentication.
+        /// </summary>
+        /// <param name="context">Contains information about the current image request and parsed commands.</param>
+        /// <param name="handling">The command collection handling.</param>
+        /// <returns>The computed HMAC.</returns>
+        internal async Task<string> ComputeHMACAsync(ImageCommandContext context, CommandHandling handling)
+        {
+            byte[] secret = this.options.HMACSecretKey;
+            if (secret is null || secret.Length == 0)
+            {
+                return null;
+            }
+
+            CommandCollection commands = this.requestParser.ParseRequestCommands(context.Context);
+            if (handling == CommandHandling.Sanitize)
+            {
+                this.StripUnknownCommands(commands);
+            }
+
```

Comment:
This can be removed, as these commands aren't used (they are already parsed/stripped and available on `context.Commands`:
```suggestion
```